cmd/minishift/cmd/start.go

```diff
@@ -160,6 +160,15 @@ func runStart(cmd *cobra.Command, args []string) {
 	}
 
 	if !isRestart {
+		// Run the default addons if MINISHIFT_HOME dir does not exist or empty
```

Comment:
> Otherwise, for any command, the installation of default addons will occur.

This is what we want, right? The idea is that you write the marker and then on execution of _any_ Minishift command we handle the post update commands. What if the user wants to "explore" any potential new add-ons via the 'minishift addons list' or maybe enable a new add-on prior to calling start. The whole idea of utilising the pre-run in root.go is to make sure that any outstanding changes we want to make as part of the update are executing as soon as possible. Does this make sense now?